### PR TITLE
feat(tokens): add support for password change and reset event

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes https://github.com/mozilla/fxa-oauth-server/issues/481
Fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1374032

### CODE REVIEW

**Comment by vladikoff** [comment]
@rfk @vbudhram r?

File: lib/db/mysql/index.js [hunk 1]
```diff
@@ -606,6 +610,23 @@ MysqlStore.prototype = {
...
+   * @param userId
+   * @returns {Promise}
+   */
+  removePublicAndCanGrantTokens: function removePublicAndCanGrantTokens(userId) {
+    const uid = buf(userId);
+
+    return this._read(QUERY_PUBLIC_CLIENTS_LIST).then((_clients) => {
+      const clientIds = _clients.map((client) => client.id);
+
+      return this._write(QUERY_DELETE_ACCESS_TOKEN_FOR_PUBLIC_CLIENTS, [uid, clientIds])
```
Comment (rfk) [inline lib/db/mysql/index.js]: Looking up the client ids in a separate query is fine, but you could skip a round-trip by using MySQL's neat ability to delete through a join:

  https://stackoverflow.com/questions/652770/delete-with-join-in-mysql#4192849

It'd look something like (untested):

```
DELETE tokens
FROM tokens INNER JOIN clients ON tokens.clientId = clients.id
WHERE tokens.userId = ? AND (clients.publicClient = 1 OR clients.canGrant = 1)
```

File: test/db/index.js [hunk 2]
```diff
@@ -397,6 +397,100 @@ describe('db', function() {
...
+    it('revokes tokens for publicClient', () => {
+      return testRemovalWithClient({
+        publicClient: true
+      }).then((tokens) => {
+        assert.equal(tokens[0], undefined);
+        assert.equal(tokens[1], undefined);
+      });
+    });
+
+    it('does not revokesTokens for not canGrant or not publicClient', () => {
```
Comment (rfk) [inline test/db/index.js]: nit: s/or/and/ ?

File: lib/db/mysql/index.js (hunk 1 above)
Comment (vladikoff) [reply lib/db/mysql/index.js]: > MySQL's neat ability

Let's stay away from that 🤕  IMHO

**Pull request merged**